crates/da-clients/celestia/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+pub mod config;
+pub mod error;
+
+use async_trait::async_trait;
+use color_eyre::Result;
+use config::CelestiaDaConfig;
+
+use celestia_rpc::{BlobClient, Client};
+use celestia_types::blob::GasPrice;
+use celestia_types::{nmt::Namespace, Blob};
+
+use da_client_interface::{DaClient, DaVerificationStatus};
+
+pub struct CelestiaDaClient {
+    client: Client,
+    nid: Namespace,
+}
+
+pub struct CelestiaDaConfigAndClient {
+    pub config: CelestiaDaConfig,
+    pub client: Client,
+}
+
+#[async_trait]
+impl DaClient for CelestiaDaClient {
+    async fn publish_state_diff(&self, state_diff: Vec<Vec<u8>>, to: &[u8; 32]) -> Result<String> {
+        // Convert the state_diffs into Blobs
+        let blobs: Result<Vec<Blob>, _> =
+            state_diff.into_iter().map(|blob_data| Blob::new(self.nid, blob_data)).collect();
+
+        // Submit the blobs to celestia
+        let height = self.client.blob_submit(blobs?.as_slice(), GasPrice::default()).await?;
+
+        println!("{}", height);
```

Comment:
avoid printlns. you can make this log::trace with proper messaging if you like

Reply:
Noted.